modules/core/04-channel/keeper/events.go

```diff
@@ -146,9 +146,15 @@ func emitSendPacketEvent(ctx sdk.Context, packet types.Packet, channel types.Cha
 	})
 }
 
-// emitRecvPacketEvent emits a receive packet event. It will be emitted both the first time a packet
+// EmitRecvPacketEvent emits a receive packet event. It will be emitted both the first time a packet
 // is received for a certain sequence and for all duplicate receives.
-func emitRecvPacketEvent(ctx sdk.Context, packet types.Packet, channel types.Channel) {
+func EmitRecvPacketEvent(ctx sdk.Context, packet types.Packet, channel *types.Channel) {
+	// An event emitted from eureka handler will not have a channel. Default initialize one
+	// with ordering set to UNORDERED.
+	if channel == nil {
+		channel = &types.Channel{Ordering: types.UNORDERED}
```

Comment:
Doesn't this also need connection hops or it will panic on trying to get the first element. Maybe we should just create a single element field with the underlying client ID.